packages/kbn-health-gateway-server/src/kibana/routes/root.ts

```diff
@@ -6,164 +6,161 @@
-
-export function createRootRoute({ config, logger }: RootRouteDependencies) {
-  const kibanaConfig = new KibanaConfig({ config, logger });
-  const fetch = configureFetch(kibanaConfig);
-
-  return {
-    method: 'GET',
-    path: GATEWAY_ROOT_ROUTE,
-    handler: async (req: Request, h: ResponseToolkit) => {
-      const responses = await fetchKibanaRoots({ fetch, kibanaConfig, logger });
-      const { body, statusCode } = mergeResponses(responses);
-      logger.debug(`Returning ${statusCode} response with body: ${JSON.stringify(body)}`);
-
-      return h.response(body).type('application/json').code(statusCode);
-    },
-  };
+interface HostStatus {
+  host: string;
+  status: Status;
+  code?: number;
+  message?: string;
 }
 
-async function fetchKibanaRoots({
-  fetch,
-  kibanaConfig,
-  logger,
-}: {
-  fetch: Fetch;
-  kibanaConfig: KibanaConfig;
-  logger: Logger;
-}) {
-  const responses = await Promise.allSettled(
-    kibanaConfig.hosts.map(async (host) => {
-      logger.debug(`Fetching response from ${host}${KIBANA_ROOT_ROUTE}`);
-      return fetch(`${host}${KIBANA_ROOT_ROUTE}`);
-    })
-  );
-
-  responses.forEach((response, index) => {
-    const host = `${kibanaConfig.hosts[index]}${KIBANA_ROOT_ROUTE}`;
-
-    if (response.status !== 'rejected') {
-      logger.debug(`Got response from ${host}: ${JSON.stringify(response.value.status)}`);
-
-      return;
-    }
+export class RootRoute implements ServerRoute {
+  private static isHealthy(response: Response) {
+    return RootRoute.isSuccess(response) || RootRoute.isUnauthorized(response);
+  }
 
-    if (response.reason instanceof Error) {
-      logger.error(response.reason);
-    }
+  private static isUnauthorized({ status, headers }: Response): boolean {
+    return status === 401 && headers.has('www-authenticate');
+  }
 
-    if (response.reason instanceof Error && response.reason.name === 'AbortError') {
-      logger.error(`Request timeout for ${host}`);
+  private static isSuccess({ status }: Response): boolean {
+    return (status >= 200 && status <= 299) || status === 302;
+  }
 
-      return;
-    }
+  private static readonly POLL_ROUTE = '/';
+  private static readonly STATUS_CODE: Record<Status, number> = {
+    healthy: 200,
+    unhealthy: 503,
+    failure: 502,
+    timeout: 504,
+  };
 
-    logger.error(
-      `No response from ${host}: ${
-        response.reason instanceof Error ? response.reason.message : JSON.stringify(response.reason)
-      }`
-    );
-  });
+  readonly method = 'GET';
+  readonly path = '/';
 
-  return responses;
-}
+  constructor(private kibanaConfig: KibanaConfig, private logger: Logger) {
+    this.handler = this.handler.bind(this);
 
-function mergeResponses(
-  responses: Array<PromiseFulfilledResult<Response> | PromiseRejectedResult>
-) {
-  const hasUnhealthyResponse = responses.some(isUnhealthyResponse);
+    return pick(this, ['method', 'path', 'handler']) as RootRoute;
```

Comment:
That's because `hapi` complains about private methods that are not in their [config schema](https://github.com/hapijs/hapi/blob/master/lib/config.js#L307).